pkg/api/node_api.go

```diff
@@ -605,19 +605,43 @@ func (a *NodeApi) nodeProcesses(w http.ResponseWriter, _ *http.Request) error {
 	return nil
 }
 
+func (a *NodeApi) stateHashDebug(height proto.Height) (*proto.StateHashJSDebug, error) {
+	stateHash, err := a.state.StateHashAtHeight(height)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get state hash at height %d", height)
+	}
+	stateHashDebug := proto.NewStateHashJSDebug(*stateHash, height, a.app.version().Version)
+	return &stateHashDebug, nil
+}
+
 func (a *NodeApi) stateHash(w http.ResponseWriter, r *http.Request) error {
 	s := chi.URLParam(r, "height")
 	height, err := strconv.ParseUint(s, 10, 64)
 	if err != nil {
 		// TODO(nickeskov): which error it should send?
 		return &BadRequestError{err}
 	}
+	stateHashDebug, err := a.stateHashDebug(height)
+	if err != nil {
+		return errors.Wrap(err, "failed to get state hash debug")
+	}
 
-	stateHash, err := a.state.StateHashAtHeight(height)
+	if err := trySendJson(w, stateHashDebug); err != nil {
+		return errors.Wrap(err, "stateHash")
+	}
+	return nil
+}
+
+func (a *NodeApi) stateHashLast(w http.ResponseWriter, _ *http.Request) error {
+	height, err := a.state.Height()
```

Comment:
Please reduce the height by 1 with corresponding checks. Scala node doesn't return state hash for the last block (liquid block as they call it) so they return last state hast at height-1.